src/synthesized_datasets/_dtypes.py

```diff
@@ -0,0 +1,58 @@
+from enum import Enum
+
+import pyspark.sql.types as st
+
+
+class DType(str, Enum):
+    """Class to define the internal dtypes that can be handled."""
+
+    BOOL = "bool"
+    NULLABLE_BOOL = "boolean"
+    DATETIME = "datetime64[ns]"
+    DATE = "date"
+    FLOAT = "float32"
+    DOUBLE = "float64"
+    INTEGER = "int32"
+    LONG = "int64"
+    NULLABLE_LONG = "Int64"
+    STRING = "string"
+    TIMEDELTA = "timedelta64[ns]"
+
+
+def create_pyspark_schema(schema: dict[str, DType]) -> st.StructType:
+    """Creates a PySpark schema from a dictionary of column names and d"""
+    return st.StructType(
+        [
+            st.StructField(name, _PS_DTYPE_MAP[dtype], _PS_NULLABLE_MAP[dtype])
+            for name, dtype in schema.items()
+        ]
+    )
+
+
+_PS_DTYPE_MAP = {
+    DType.BOOL: st.BooleanType(),
+    DType.NULLABLE_BOOL: st.BooleanType(),
+    DType.DATETIME: st.TimestampType(),
+    DType.DATE: st.DateType(),
+    DType.FLOAT: st.FloatType(),
+    DType.DOUBLE: st.DoubleType(),
+    DType.INTEGER: st.IntegerType(),
+    DType.LONG: st.LongType(),
+    DType.NULLABLE_LONG: st.FloatType(),
```

Comment:
should this be st.LongType() ?

Reply:
Maybe with an example I can make more sense:

Say you have a column a table with the following columns:

```csv
# some csv file
colA,colB,colC
0,1,1.0
1,,
3,2,2.0
0,1,1.0
```

Both `colA` and `colB` can be read into spark as integers (one being non-null and the other being nullable. But Spark can't read `colC` as an integer column. Well... it does, but it says everything is Null. i.e.,
```
+----+----+----+
|colA|colB|colC|
+----+----+----+
|   0|   1| NaN|  
|   1|    |    | 
|   3|   2| NaN| 
|   0|   1| NaN| 
+----+----+----+
```
So whilst pandas can handle the decimal points, spark cannot. The mapping of Nullable_long to FloatType simply means spark creates the following dataframe which I think is good enough for now! (I think we may want to ultimately rewrite the file)

```
+----+----+----+
|colA|colB|colC|
+----+----+----+
|   0|   1| 1.0|  
|   1|    |    | 
|   3|   2| 2.0| 
|   0|   1| 1.0| 
+----+----+----+
```